pkg/distributor/distributorerror/error.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import "github.com/grafana/mimir/pkg/util/validation"
+
+type Error interface {
+	error
+	DistributorError()
+}
+
+// DistributorPushError is a generic error returned on
+// distributor's write path.
+type DistributorPushError struct {
+	err error
+}
+
+// Error makes DistributorPushError implement error interface.
+func (e DistributorPushError) Error() string {
+	return e.err.Error()
+}
+
+// Unwrap makes DistributorPushError implement Wrapper interface.
+func (e DistributorPushError) Unwrap() error {
+	return e.err
+}
+
+// PushError makes DistributorPushError implement pusherror.Error interface.
+func (e DistributorPushError) PushError() {}
+
+// Error makes DistributorPushError implement Error interface.
+func (e DistributorPushError) DistributorError() {}
+
+// NewDistributorPushError wraps the given error into a DistributorPushError.
+func NewDistributorPushError(err error) DistributorPushError {
+	return DistributorPushError{
+		err: err,
+	}
+}
+
+// ReplicasNotMatchDistributorPushError is an implementation of Error,
+// corresponding to distributor.replicasNotMatchError.
+type ReplicasNotMatchDistributorPushError struct {
```

Comment:
I have replaced the `DistributorPushError` suffix with `Error` suffix.

Reply:
As long as the package is called `distributorerror`, the `Error` suffix is also redundant, because the full name is:
`distributorerror.ReplicasNotMatchDistributorPush`.